nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -1045,7 +1045,8 @@ in
 
     boot.initrd.availableKernelModules =
       optional (cfg.qemu.diskInterface == "scsi") "sym53c8xx"
-      ++ optional (cfg.tpm.enable) "tpm_tis";
+      ++ optional (cfg.tpm.enable) "tpm_tis"
+      ++ optional  (cfg.sharedDirectories != {}) "9pnet_virtio";
```

Comment:
We already do this in the guest profile: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/profiles/qemu-guest.nix

Reply:
Ah, fair point. I'll fix, test, and merge.